PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py

```diff
@@ -1641,12 +1641,13 @@ def ak4JetReclustering(self,process, pfCandCollection, patMetModuleSequence, pos
             from RecoJets.JetProducers.ak4PFJets_cfi import ak4PFJets
             #if chs:
             addToProcessAndTask(jetColName, ak4PFJets.clone(), process, task)
-            getattr(process, jetColName).src = pfCandColl 
+            getattr(process, jetColName).src = pfCandColl
             getattr(process, jetColName).doAreaFastjet = True
 
             #puppi
             if self._parameters["Puppi"].value:
                 getattr(process, jetColName).src = cms.InputTag("puppi")
+                getattr(process, jetColName).applyWeights = True
```

Comment:
https://github.com/cms-sw/cmssw/pull/31927/files?diff=unified&w=1#diff-79860e2d37e9cc0867e9de61faab9b79e0283ebf986856b7a05f517e6d1d083aL1640-R1640

-                getattr(process, jetColName).applyWeights = True
+               getattr(process, jetColName).applyWeight = True

Reply:
isn't this `applyWeight` a recent feature related to puppi moved to running in reco, specific to 11_X?

Reply:
> isn't this `applyWeight` a recent feature related to puppi moved to running in reco, specific to 11_X?

Indeed: (besides the wrong spelling pointed out by Maria) this `applyWeight` was introduced by #28936 and #29254, never backported to 10_6

Reply:
@slava77 @perrotta 

puppiForMET changed a lot between 11_2 and 10_6 not only in RECO, i.e. the function  makePuppiesFromMiniAOD

https://github.com/cms-sw/cmssw/blob/CMSSW_11_2_X/PhysicsTools/PatAlgos/python/slimming/puppiForMET_cff.py#L13-L25

https://github.com/cms-sw/cmssw/blob/CMSSW_10_6_X/PhysicsTools/PatAlgos/python/slimming/puppiForMET_cff.py#L37-L65

based on your code review, do you have some suggestion from where we should look ?

@ahinzmann

Reply:
@mariadalfonso from where you should look for what?

Our review simply says that the features ruled by the parameter `applyWeight` do not exhist in 10_6. Either you can get rid of them, and therefore remove all modifications that you introducded for PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py as implemented in this PR; or they are necessary ingredients of this backport, and therefore you must manage to carefully import all related updates now integrated in the master.

I pointed out two PRs which introduce those features in reco: as you say, there were a few additional ones which further updated puppiForMET for nanoAOD. If you really want to backport all them you must be extremely careful of not changing the default outputs in a stable production release, and to make them compliant with the planned re-miniAOD campaign. I suspect it can be a though work, and then you should evaluate what do you really need for 10_6

Reply:
@perrotta 
yes, I agree that the `applyWeight` is not needed, thanks for the clarification, Alejandro will remove from the PR; 
let's hope that ahinzmann suggestion solve the problem